biz.aQute.bndlib/src/aQute/bnd/build/Workspace.java

```diff
@@ -121,13 +123,22 @@
 	static final int			BUFFER_SIZE						= IOConstants.PAGE_SIZE * 16;
 	private static final String	PLUGIN_STANDALONE		= Constants.PLUGIN + ".standalone_";
 
-	class WorkspaceData {
-		List<RepositoryPlugin>	repositories;
-		RemoteWorkspaceServer	remoteServer;
-		Lazy<WorkspaceClassIndex>				classIndex	= new Lazy<>(
-			() -> new WorkspaceClassIndex(Workspace.this));
-		Lazy<WorkspaceExternalPluginHandler>	externalPlugins	= new Lazy<>(
-			() -> new WorkspaceExternalPluginHandler(Workspace.this));
+	class WorkspaceData implements Closeable {
+		final Supplier<List<RepositoryPlugin>>			repositories;
+		volatile RemoteWorkspaceServer					remoteServer;
+		final Supplier<WorkspaceClassIndex>				classIndex;
+		final Supplier<WorkspaceExternalPluginHandler>	externalPlugins;
+
+		WorkspaceData() {
+			repositories = Memoize.supplier(Workspace.this::initRepositories);
```

Comment:
I am not happy with the replacement of Lazy with Memoize. It has no close support nor introspective facilities. Can  you explain why you made this change?

Reply:
I would like to revert until there is a compelling reason to make it less functional and more complex?

Reply:
Lazy has some serious issues. I am happy to discuss when you are available.